File: AspNetReactMySql/AppBackEnd/Models/Book.cs
using System.ComponentModel.DataAnnotations;

namespace AppBackEnd.Models
{
public class Book
    {
        [KeyAttribute]
        public int Id { get; set; }
        [Required]
        public string Name { get; set; }
        public string ISBN {get; set; }
        [Required]
        public string Author { get; set; }
        [Required]
        public string Description { get; set; }
        public int Year { get; set; }
        public string ImageUrl { get; set; }        
        public string Publisher { get; set; }
    }
}

File: src/Books/Book.js
import React, { useState } from 'react';
import './Book.scss'
function Book({prop}) {
    
    const [name, setName] = useState(prop.title);
    const [author, setAuthor] = useState(prop.author);
    const [year, setYear] = useState(prop.year);
    const [image, setImage] = useState(prop.imageUrl);
    const [isbn, setIsbn] = useState(prop.isbn);
    const [desc, setDesc] = useState(prop.description);
    return(
        <div className='book'>
            <img src={image} alt="image" />
            <div className='text'>
                <h2>{name} ({year})</h2>
                <h3>{author}</h3>
                <h4>ISBN: {isbn}</h4>

                <p>{desc.substring(0,190)}</p>
            </div>
            
        </div>
    );
}

export default Book;
